src/worker.js

```diff
@@ -274,3 +264,57 @@ this.onmessage = function(e) {
     throw e;
   }
 };
+
+#if ENVIRONMENT_MAY_BE_NODE
+// Node.js support
+if (typeof process === 'object' && typeof process.versions === 'object' && typeof process.versions.node === 'string') {
+  // Create as web-worker-like an environment as we can.
+  self = {
+    location: {
+      href: __filename
+    }
+  };
+
+  var onmessage = this.onmessage;
+
+  var nodeWorkerThreads = require('worker_threads');
```

Comment:
This will throw on versions of Node.js that don’t have this module – is that okay?

Reply:
Good point, yeah, I'll make it show an error that likely a newer node.js is needed if that module is missing.